### ISSUE TITLE
Customise the trustedCall response mode

### ISSUE THREAD

**Issue by Kailai-Wang** [litentry/litentry-parachain#1922]
### Context

A follow-up issue of #1883
Now that we start to support both synchronous and asynchronous responses, we might want to discuss and decide which mode to use in practice.

There're a few options:

1. Try to detect it (e.g. by signer) and automatically select the mode
2. Always send both and it's up to the client which one to use **=> current implementation**
3. Have a per-trustedCall configuration and follow it in the runtime, e.g.
    - `set_user_shielding_key` => DI response via ws
     - `link_identity` => II response via parachain
     - `request-vc` => both DI and II responses

cc @jonalvarezz @kaylawangnow

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

**Comment by kziemianek** [comment]
The `StfExecutor` already knows which calls are submitted directly/indirectly so with this information we can conditionally send II/DI responses.

It means:
- for indirect invocations the node will send extrinsics back to parachain signaling processing result
- for direct invocations the node will send rpc responses back to client, but will not send extrinsics back to parachain
- if there is any state synchronization needed (for example after VC generation) it will be performed as it is

I can tackle it once #1957 is merged.

**Comment by Kailai-Wang** [comment]
> for direct invocations the node will send rpc responses back to client, but will not send extrinsics back to parachain

Hold on about this one - for some request we might want both, or at least it will have both synchronous responses and parachain effects

**Comment by kziemianek** [comment]
@Kailai-Wang which case do you mean ? Isn't it covered by third dot (case `vc_issued` ) ?
